go/pserver/service.go

```diff
@@ -44,14 +45,16 @@ type Gradient Parameter
 type Service struct {
 	initialized chan struct{}
 
-	mu       sync.Mutex
-	opt      *optimizer
+	mu sync.Mutex
+	// injection from parameter to optimizer
+	optMap   map[string]*optimizer
```

Comment:
If we already have `optMap` and Optimizer owns the parameter memory, we no longer need `paramMap`.